airbyte-integrations/connectors/source-shopify/source_shopify/utils.py

```diff
@@ -108,3 +109,53 @@ def wrapper_balance_rate_limit(*args, **kwargs):
             return wrapper_balance_rate_limit
 
         return decorator
+
+
+class BackupStreamState:
```

Comment:
nit: can we change this to `EagerlyCachedStreamState`?

Backup sounds like we are backing something up. Here, what we are doing is making sure we eagerly using already loaded state.

Reply:
Corrected. Thank you)